### ISSUE TITLE
Fix flake8 rule B024

### ISSUE THREAD

**Issue by MortalHappiness** [ray-project/ray#48069]
This is a subtask of https://github.com/ray-project/ray/issues/47991. See the parent issue for more information.

https://github.com/PyCQA/flake8-bugbear

**Comment by win5923** [comment]
hi @MortalHappiness, Does this issue still need to be resolved? If so, may I take it on?

**Comment by MortalHappiness** [comment]
@win5923 Definitely! By the way, if you are interested in https://github.com/ray-project/ray/issues/48508 you can also take it. It's more important than fixing those rules now.

**Comment by MortalHappiness** [comment]
Close this because we decide to migrate from `flake8` to `ruff` instead of upgrading `flake8`.
